cylc/flow/suite_files.py

```diff
@@ -926,4 +935,305 @@ def get_cylc_run_abs_path(path):
+        no_symlinks (bool): Flag as True to skip making localhost symlink dirs
+
+    Return:
+        source (Path): The source direcory.
+        rundir (Path): The directory the workflow has been installed into.
+        flow_name (str): The installed suite name (which may be computed here).
+
+    Raise:
+        WorkflowFilesError:
+            No flow.cylc file found in source location.
+            Illegal name (can look like a relative path, but not absolute).
+            Another suite already has this name (unless --redirect).
+            Trying to install a workflow that is nested inside of another.
+    """
+
+    if not source:
+        source = Path.cwd()
+    elif Path(source).name == SuiteFiles.FLOW_FILE:
+        source = Path(source).parent
+    source = Path(source).expanduser()
+    if not flow_name:
+        flow_name = (Path.cwd().stem)
+    validate_flow_name(flow_name)
+    if run_name == '_cylc-install':
+        raise WorkflowFilesError(
+            'Run name cannot be "_cylc-install".'
+            ' Please choose another run name.')
+    validate_source_dir(source, flow_name)
+    run_path_base = Path(get_workflow_run_dir(flow_name)).expanduser()
+    relink, run_num, rundir = get_run_dir(run_path_base, run_name, no_run_name)
+    if Path(rundir).exists():
+        raise WorkflowFilesError(
+            f"\"{rundir}\" exists."
+            " Try using cylc reinstall. Alternatively, install with another"
+            " name, using the --run-name option.")
+    check_nested_run_dirs(rundir, flow_name)
+    symlinks_created = {}
+    if not no_symlinks:
+        sub_dir = flow_name
+        if run_num:
+            sub_dir += '/' + f'run{run_num}'
+        symlinks_created = make_localhost_symlinks(rundir, sub_dir)
+    _open_install_log(rundir)
+    if not no_symlinks and bool(symlinks_created) is True:
+        for src, dst in symlinks_created.items():
+            INSTALL_LOG.info(f"Symlink created from {src} to {dst}")
+    try:
+        rundir.mkdir(exist_ok=True)
+    except OSError as e:
+        if e.strerror == "File exists":
+            raise WorkflowFilesError(f"Run directory already exists : {e}")
+    if relink:
+        link_runN(rundir)
+    create_workflow_srv_dir(rundir)
+    # flow.cylc must exist so we can detect accidentally reversed args.
+    flow_file_path = source.joinpath(SuiteFiles.FLOW_FILE)
+    if not flow_file_path.is_file():
+        # If using deprecated suite.rc, symlink it into flow.cylc:
+        suite_rc_path = source.joinpath(SuiteFiles.SUITE_RC)
+        if suite_rc_path.is_file():
+            flow_file_path.symlink_to(suite_rc_path)
+            INSTALL_LOG.warning(
+                f'The filename "{SuiteFiles.SUITE_RC}" is deprecated in favour'
+                f' of "{SuiteFiles.FLOW_FILE}". Symlink created.')
+        else:
+            raise WorkflowFilesError(
+                f'no {SuiteFiles.FLOW_FILE} or {SuiteFiles.SUITE_RC}'
+                f' in {source}')
```

Comment:
I think this logic is a duplicate of the `parse_suite_arg`, factor into another function?

Reply:
Thanks, I've refactored.